### ISSUE TITLE
false positive invalid tag JavadocType 

### ISSUE THREAD

**Issue by Machine-Maker** [checkstyle/checkstyle#16087]
```bash
/var/tmp $ javac Test.java
# success

/var/tmp $ cat checkstyle.xml
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="JavadocType"/>
  </module>
</module>

/var/tmp $ cat Test.java
/**
 * <pre>{@code
 * @Internal
 * class Other {
 *   @Nullable String test(String param) {
 *   }
 * }
 * }</pre>
 */
public class Test {
  public void test(String param) {
  }
}

/var/tmp $ java -jar checkstyle-10.21.0-all.jar -c checkstyle.xml test.java
Starting audit...
[ERROR] /Users/jake/Projects/Java/checkstyle-test/Test.java:3:4: Unknown tag 'Internal'. [JavadocType]
[ERROR] /Users/jake/Projects/Java/checkstyle-test/Test.java:5:6: Unknown tag 'Nullable'. [JavadocType]
Audit done.
Checkstyle ends with 2 errors.
```

---

`@Internal` and `@Nullable` should not be reported as an unknown tag. It is inside a valid pre/code tag combo which is displayed correctly by the javadoc tool. You can see the rendered html below.
![CleanShot 2024-12-29 at 07 39 32@2x](https://github.com/user-attachments/assets/93a9c094-24fe-42e5-a62d-871b3545707d)

**Comment by romani** [comment]
All logic is at https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java#L154

It is Check, old Check, that doesn't use AST.
So fixing issue on it might be not easy. 

But issue is valid, no violations are expected.

**Comment by Machine-Maker** [comment]
For anyone else who finds this, I solved it with the following suppression filter
```xml
    <module name="SuppressionSingleFilter">
        <property name="checks" value="Javadoc(Method|Type)"/>
        <!--[A-Z].* check is for java annotations inside code blocks inside javadocs-->
        <property name="message" value="^Unknown tag '(${custom_javadoc_tags}|[A-Z].*)'.$"/>
    </module>
```
`${custom_javadocs_tags}` is an injected property with all the custom javadoc tags registered with the javadoc tool in gradle (in my case, just `apiNote`). But after that, the `[A-Z].*` regex, just suppresses warnings if the first letter of the tag is capital (which isn't the case for any real javadoc tag and is the case for most (normal) java annotations).

**Comment by XN137** [comment]
hitting a similar issue after upgrading from `10.20.0` to `10.21.1`:
https://github.com/checkstyle/checkstyle/issues/14573#issuecomment-2575620325

it seems like content inside `<pre>..</pre>` isnt ignored (but should be?).

**Comment by romani** [comment]
It should be. 
Main rule: if javadoc passed/generated by jdk tool `javadoc` , so checkstyle should  support such javadoc also.

**Comment by YuktiNandwana** [comment]
Hello @romani ,  I would like to work on this issue and prepare a fix. Please let me know if that works.

**Comment by romani** [comment]
@YuktiNandwana , this is complicated issue, please choose next "good xxxx issue" it will help to be onboarded smoothly , without frustration. This issue will wait for you :).

**Comment by YuktiNandwana** [comment]
Thanks for the guidance, @romani 🙏  
I understand this issue is complex for onboarding.  
I’ll pick another approved “good first issue” and come back to this later with more experience.

**Comment by aclfe** [comment]
Hi @romani, since Yukti has stepped aside, I would like to pick this up. 

​This bug seems vaguely familiar to the JavadocUtilizingSpace check that I'm working with. It also has similar requirements to follow. The difference being AST as you mentioned. 

I'm on it.

**Comment by romani** [comment]
fix is merged

**Comment by snuyanzin** [comment]
hi @romani we (Apache Flink) faced same issue
we can not bump to 13.1.0 since we have to be sure the project is able to be built with jdk11, 17, 21
can it be ported to 10.x.x please?

**Comment by romani** [comment]
Unfortunately , we are tiny team, we have no power to do back porting.
You probably should update your build script to do checkstyle only on jdk21 runtime.

As your project is open source, What we can do improve for future and make a "partnership": you  come to state where all checkstyle violations fails a build, we add your project to our CI pipeline to validate regression on each commit. 
Example on how recently we did with Kafka 
https://github.com/checkstyle/checkstyle/issues/16003
If you are interested, we can discuss details in separate new issue.

As we put it in our pipeline, checkstyle contributors will send updates to your repository to fix new violations that will happen in next release.
